targets/targets.json

```diff
@@ -1057,326 +1009,270 @@
-        "device_has": ["ANALOGIN", "ANALOGOUT", "CAN", "I2C", "I2CSLAVE", "I2C_ASYNCH", "INTERRUPTIN", "LOWPOWERTIMER", "PORTIN", "PORTINOUT", "PORTOUT", "PWMOUT", "RTC", "SERIAL", "SERIAL_ASYNCH", "SLEEP", "SPI", "SPISLAVE", "SPI_ASYNCH", "STDIO_MESSAGES", "TRNG"],
+        "device_has_add": ["ANALOGOUT", "CAN", "LOWPOWERTIMER", "SERIAL_ASYNCH", "TRNG"],
         "features": ["LWIP"],
         "release_versions": ["2", "5"],
         "device_name": "STM32F769NI"
     },
     "DISCO_L475VG_IOT01A": {
-        "inherits": ["Target"],
+        "inherits": ["FAMILY_STM32"],
         "core": "Cortex-M4F",
-        "extra_labels": ["STM", "STM32L4", "STM32L475xG", "STM32L475VG"],
-        "supported_toolchains": ["ARM", "uARM", "IAR", "GCC_ARM"],
-        "default_toolchain": "ARM",
+        "extra_labels_add": ["STM32L4", "STM32L475xG", "STM32L475VG"],
         "supported_form_factors": ["ARDUINO"],
         "detect_code": ["0764"],
-        "macros": ["TRANSACTION_QUEUE_SIZE_SPI=2", "USBHOST_OTHER"],
-        "device_has": ["ANALOGIN", "ANALOGOUT", "CAN", "I2C", "I2CSLAVE", "I2C_ASYNCH", "INTERRUPTIN", "LOWPOWERTIMER", "PORTIN", "PORTINOUT", "PORTOUT", "PWMOUT", "RTC", "SERIAL", "SERIAL_FC", "SLEEP", "SPI", "SPISLAVE", "SPI_ASYNCH", "STDIO_MESSAGES", "TRNG"],
+        "macros_add": ["USBHOST_OTHER"],
+        "device_has_add": ["ANALOGOUT", "CAN", "LOWPOWERTIMER", "SERIAL_FC", "TRNG"],
         "release_versions": ["2", "5"],
         "device_name": "STM32L475VG"
     },
     "DISCO_L476VG": {
-        "inherits": ["Target"],
+        "inherits": ["FAMILY_STM32"],
         "core": "Cortex-M4F",
-        "default_toolchain": "ARM",
-        "extra_labels": ["STM", "STM32L4", "STM32L476xG", "STM32L476VG"],
-        "supported_toolchains": ["ARM", "uARM", "IAR", "GCC_ARM"],
+        "extra_labels_add": ["STM32L4", "STM32L476xG", "STM32L476VG"],
         "detect_code": ["0820"],
-        "macros": ["TRANSACTION_QUEUE_SIZE_SPI=2", "USBHOST_OTHER"],
-        "device_has": ["ANALOGIN", "ANALOGOUT", "CAN", "I2C", "I2CSLAVE", "I2C_ASYNCH", "INTERRUPTIN", "LOWPOWERTIMER", "PORTIN", "PORTINOUT", "PORTOUT", "PWMOUT", "RTC", "SERIAL", "SERIAL_FC", "SLEEP", "SPI", "SPISLAVE", "SPI_ASYNCH", "STDIO_MESSAGES", "TRNG", "FLASH"],
+        "macros_add": ["USBHOST_OTHER"],
+        "device_has_add": ["ANALOGOUT", "CAN", "LOWPOWERTIMER", "SERIAL_FC", "TRNG", "FLASH"],
         "release_versions": ["2", "5"],
         "device_name": "STM32L476VG"
     },
     "MTS_MDOT_F405RG": {
-        "inherits": ["Target"],
+        "inherits": ["FAMILY_STM32"],
         "core": "Cortex-M4F",
-        "supported_toolchains": ["ARM", "uARM", "GCC_ARM", "IAR"],
-        "extra_labels": ["STM", "STM32F4", "STM32F405RG"],
+        "extra_labels_add": ["STM32F4", "STM32F405RG"],
         "is_disk_virtual": true,
-        "macros": ["HSE_VALUE=26000000", "TRANSACTION_QUEUE_SIZE_SPI=2"],
+        "macros_add": ["HSE_VALUE=26000000"],
         "progen": {"target": "mts-mdot-f405rg"},
```

Comment:
I know that you did not change this, but could you get rid of this progen key? It's not used anymore.